File: Java/Intervals/InsertIntervals.java
// https://leetcode.com/problems/insert-interval/

class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        if (intervals == null) return new int[][] {newInterval};
        List<int[]> res = new ArrayList<> ();
        for (int[] interval: intervals) {
            if (interval[1] < newInterval[0]) {
                res.add(interval);
            }
            else if (interval[0] > newInterval[1]) {
                res.add(newInterval);
                newInterval = interval;
            }
            else {
                newInterval[0] = Math.min(newInterval[0], interval[0]);
                newInterval[1] = Math.max(newInterval[1], interval[1]);
            }
        }
        res.add(newInterval);
        return res.toArray(new int[res.size()][]);
        
        
    }
}

File: Python/Intervals/InsertIntervals.py
# https://leetcode.com/problems/insert-interval/

class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        for index, interval in enumerate(intervals):
            if interval[1] < newInterval[0]:
                res.append(interval)
            elif interval[0] > newInterval[1]:
                res.append(newInterval)
                return res + intervals[index:]
            else:
                newInterval[0] = min(interval[0], newInterval[0])
                newInterval[1] = max(interval[1], newInterval[1])
        res.append(newInterval)
        return res
        
